### PR TITLE
Serve Directories

### PR DESCRIPTION
**Pull request by damonkelley**
This adds the ability to serve a directory. The directory view is a list of links to each file or subdirectory in the directory.

Files and directories are being served through a custom fallback request handler, which is defined inside of the cob-spec app and injected into the `route` function. If the requested path does not match one of the routes, the fallback will first check if the path is an existing directory, then if it is an existing file, otherwise it will respond with `404`

Also, the cob-spec application now accepts command line options for the port and directory to serve from. If none are specified, it looks for `./public` and uses port 5000.

Story 25
### Preview

image
### Cob Spec running against these changes

image

### CODE REVIEW

**Comment by trptcolin** [comment]
This is looking really good. The PR structure itself (description & screenshots) is excellent. Didn't find any issues on a cursory pass at code review. Lots of good things in the code. 

Maybe @Nayshins can find something bad? I don't see anything 👍

**Comment by Nayshins** [comment]
I am going to second @trptcolin on this one. This PR is looking pretty great!

File: spec/http_clj/router_spec.clj [hunk 1]
```diff
@@ -16,14 +18,16 @@
...
   (it "will not find the handler if the path matches but the method does not"
     (let [{status :status body :body} (router/route
                                         {:method "GET" :path "/b"} @routes)]
-      (should= 404 status)
-      (should-contain "Route not found" body))))
+      (should= 404 status)))
+
+  (it "can be provided with an alternate fallback handler"
+    (should-invoke fallback {:times 1}
+                   (router/route {:method "GET" :path "/c"} @routes :fallback fallback ))))
```
Comment (trptcolin) [inline spec/http_clj/router_spec.clj]: Trailing whitespace before closing parens here.

**Comment by trptcolin** [comment]
OK so I found one whitespace nitpick 😄

**Pull request merged**